docs/ui/Toc.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import { Link, List } from '@marigold/components';
+
+export interface TocProps {
+  data: string;
+}
+
+type Item = {
+  anchor: string;
+  title: string;
+  id: string;
+  level: string;
+};
+
+export const Toc = ({ data }: TocProps) => {
+  const elements = JSON.parse(data) as Item[];
+
+  const [, setIsMounted] = useState(false);
+
+  const ref = useRef<Element>();
+
+  const itemId = elements.map(item => item.id);
+  const activeItem = useActiveItem(itemId);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      ref.current = document.querySelector('#toc') || undefined;
+      setIsMounted(true);
```

Comment:
Is this needed? the value is never read

Reply:
Without it doesn't render. I think I did this because you have set the ref.current to the toc element... to reference to this @sebald